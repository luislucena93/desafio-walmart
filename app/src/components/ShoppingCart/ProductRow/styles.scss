@import '../../../commons/styleVars';

.productRow {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 10px;
  border-bottom: solid 1px $shadowColor;
  .addRemoveButtons {
    visibility: hidden;
    margin: 10px 0px;
    p {
      padding: 0px 5px;
    }
  }
  &:hover {
    .addRemoveButtons {
      visibility: visible;
    }
  }
  img {
    width: 100px;
  }
  p {
    margin-top: 0px;
    margin-bottom: 1px;
  }
  .productSummary {
    padding-left: 10px;
    flex-grow: 3;
  }
  .productTotal {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
  }
}