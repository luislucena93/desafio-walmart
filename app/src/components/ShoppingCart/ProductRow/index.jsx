import React from 'react';
import './styles.scss';
import FormatMoney from '../../../helpers/formatters';

const ProductRow = (props) => {
    const {product, removeProduct, addProduct} = props;
    return (
        <div className={"productRow"}>
            <img src={`http://${product.info.image}`}/>
            <div className={"productSummary"}>
                <p className={"bold"}>{product.info.brand}</p>
                <p>{product.info.description}</p>
                <p>${FormatMoney(product.info.price)}</p>
                <p className={"bold"}>Cantidad: {product.count}</p>
            </div>
            <div className={"productTotal"}>
                <p className={"bold"}>${FormatMoney(product.count * product.info.price)}</p>
                <div className={"addRemoveButtons"}>
                    <button onClick={() => removeProduct(product.info)}>-</button>
                    <p>{product.count}</p>
                    <button onClick={() => addProduct(product.info)}>+</button>
                </div>
            </div>
        </div>
    )
};

export default ProductRow;