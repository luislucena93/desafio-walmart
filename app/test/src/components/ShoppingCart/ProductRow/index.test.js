import React from 'react';
import {render, fireEvent, screen} from '@testing-library/react';
import ProductRow from "../../../../../../app/src/components/ShoppingCart/ProductRow/index";

describe('ProductRow component', () => {

    let mockAddProduct;
    let mockRemoveProduct;
    const testSelectedProduct = {
        count: 2,
        info: {id: 'testId', brand: 'testBrand', description: 'testDescription', image: 'testImage', price: 1000}
    };

    beforeEach(() => {
        mockAddProduct = jest.fn();
        mockRemoveProduct = jest.fn();
    });

    it('should render ProdcutDisplay correctly for product with quantity 0', () => {
        const {container} = render(<ProductRow product={testSelectedProduct} addProduct={mockAddProduct}
                                               removeProduct={mockRemoveProduct}/>);
        expect(container).toMatchSnapshot();
    });

    it('should call addProduct function when click the button "+"', () => {
        render(<ProductRow product={testSelectedProduct} addProduct={mockAddProduct}
                           removeProduct={mockRemoveProduct}/>);
        fireEvent.click(screen.getByText('+'));
        expect(mockAddProduct).toHaveBeenCalledWith(testSelectedProduct.info);
    })

    it('should call deleteProduct function when click the button "-"', () => {
        render(<ProductRow product={testSelectedProduct} addProduct={mockAddProduct}
                           removeProduct={mockRemoveProduct}/>);
        fireEvent.click(screen.getByText('-'));
        expect(mockRemoveProduct).toHaveBeenCalledWith(testSelectedProduct.info);
    })

});