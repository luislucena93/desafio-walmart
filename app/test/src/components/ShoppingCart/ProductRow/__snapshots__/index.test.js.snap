// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductRow component should render ProdcutDisplay correctly for product with quantity 0 1`] = `
<div>
  <div
    class="productRow"
  >
    <img
      src="http://testImage"
    />
    <div
      class="productSummary"
    >
      <p
        class="bold"
      >
        testBrand
      </p>
      <p>
        testDescription
      </p>
      <p>
        $
        1.000
      </p>
      <p
        class="bold"
      >
        Cantidad: 
        2
      </p>
    </div>
    <div
      class="productTotal"
    >
      <p
        class="bold"
      >
        $
        2.000
      </p>
      <div
        class="addRemoveButtons"
      >
        <button>
          -
        </button>
        <p>
          2
        </p>
        <button>
          +
        </button>
      </div>
    </div>
  </div>
</div>
`;
